update dashmap dependency to pick up 4.x releases

ahornby opened the pull request:
dashmap 4.x ended up as a maintenence release rather than the lockless redesign they planned so this is not super exciting, but might as well keep current

tested with cargo test

ahornby committed:
update dashmap dependency to pick up 4.x releases

dashmap 4.x ended up as a maintenence release rather than the lockless redesign they planned so this is not super exciting, but might as well keep current

tested with cargo test

ahornby commented:
From similar change on internment seems this is a bit slower.  Following up in https://github.com/xacrimon/dashmap/issues/144

ryzhyk commented:
Yes, I tried upgrading to dashmap 4 some time ago and gave up for the same reason.

Thanks for following up with maintainers!

ahornby committed:
Add ahash hasher to replicate dashmap 3.x hasher setup

Pass explicit hasher to DashMap, using same version of ahash that dashmap 3.11.10 was using

This brings ArcIntern performance with dashmap 4.x performance close to dashmap 3.x

ahornby commented:
> Yes, I tried upgrading to dashmap 4 some time ago and gave up for the same reason.
> 
> Thanks for following up with maintainers!

Added a commit that makes the dashmap 4.x performance close to dashmap 3.x (or perhaps equivalent, but my measurements are unfortunately rather noisy)

ryzhyk commented:
Thanks, @ahornby !

ryzhyk commented:
I created a crates.io release too (v0.5.1).
